jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/json/JsonCfnParser.kt

```diff
@@ -0,0 +1,395 @@
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceDependsOnNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcePropertiesNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcePropertyNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceTypeNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcesNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnRootNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnScalarValueNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnSecondLevelMappingNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnSection
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnTransformNode
+
+class JsonCfnParser private constructor() {
+    private val problems = mutableListOf<CfnProblem>()
+    private val nodeMap = BidirectionalMap<CfnNode, PsiElement>()
+
+    fun handleRoot(root: JsonObject): CfnRootNode {
+        val translatedSections = root.propertyList.mapNotNull { translateSection(it) }
+
+        return CfnRootNode(
+            translatedSections.findSection<CfnMetadataNode>(),
+            translatedSections.findSection<CfnTransformNode>(),
+            translatedSections.findSection<CfnParametersNode>(),
+            translatedSections.findSection<CfnMappingsNode>(),
+            translatedSections.findSection<CfnConditionsNode>(),
+            translatedSections.findSection<CfnResourcesNode>(),
+            translatedSections.findSection<CfnGlobalsNode>(),
+            translatedSections.findSection<CfnOutputsNode>()
+        )
+    }
+
+    private inline fun <reified T> Collection<CfnNode>.findSection(): T? = this.firstOrNull { it is T } as T?
+
+    private fun translateSection(property: JsonProperty): CfnNode? {
+        val name = property.name
+        val value = property.value
+
+        if (name.isEmpty() || value == null) {
+            return null
+        }
+
+        return when (CfnSection.fromName(name)) {
+            CfnSection.FormatVersion -> handleFormatVersion(value)
+            CfnSection.Transform -> handleTransform(property)
+            CfnSection.Parameters -> handleParameters(property)
+            CfnSection.Resources -> handleResources(property)
+            CfnSection.Conditions -> handleConditions(property)
+            CfnSection.Metadata -> handleMetadata(property)
+            CfnSection.Outputs -> handleOutputs(property)
+            CfnSection.Mappings -> handleMappings(property)
+            else -> {
+                null
+            }
+        }
+    }
+
+    private fun handleFormatVersion(value: JsonValue): CfnNode? {
+        val version = checkAndGetStringElement(value)?.value ?: return null
+
+        if (version !in SUPPORTED_VERSIONS) {
+            // TODO
```

Comment:
Should we throw for now so we don't forget to do the todo?

Reply:
Problem was added in next revision